### PR TITLE
feat: support using mwc token for fabric openai transformer

### PR DESCRIPTION
**Pull request by lhrotk**
## What changes are proposed in this pull request?

Support using fabric mwc token to auth request to internal LLM endpoints

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [x] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [x] Yes. Make sure you have added samples following below steps.

### CODE REVIEW

File: cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/CognitiveServiceBase.scala [hunk 1]
```diff
@@ -324,7 +361,8 @@ trait HasCognitiveServiceInput extends HasURL with HasSubscriptionKey with HasAA
       } else {
         val req = prepareMethod()
         req.setURI(new URI(rowToUrl(row)))
-        addHeaders(req, getValueOpt(row, subscriptionKey), getValueOpt(row, AADToken), contentType(row))
+        addHeaders(req, getValueOpt(row, subscriptionKey), getValueOpt(row, AADToken),
+          contentType(row), getValueOpt(row, CustomAuthHeader))
```
Comment (svotaw) [inline cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/CognitiveServiceBase.scala:365]: if you make a call multi-line, make one parameter per line

**Review by mhamilton723 — APPROVED** [review]

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**